api/candlepin-api-spec.yaml

```diff
@@ -22,6 +42,44 @@ components:
         requestUuid:
           type: string
 
+    StatusDTO:
+      description: Represents a deleted consumer
+      properties:
+        mode:
+          type: string
+        modeReason:
+          type: string
+        modeChangeTime:
+          type: string
+          format: date-time
+        result:
+          type: boolean
+          example: true
+        version:
+          type: string
+          example: 0.9.10
+        release:
+          type: string
+        standalone:
+          type: boolean
+        timeUTC:
+          type: string
+          format: date-time
+        rulesSource:
+          type: string
+        rulesVersion:
+          type: string
+          example: 5.8
+        capabilities:
+          type: array
+          items:
+            type: string
+        keycloakRealm:
```

Comment:
Something of note here: when we added the keycloak bits, we didn't want the values to be present in status if it was disabled (which it would be in most places); hence the separated DTOs. We should probably retain that.

Reply:
@Ceiu 
Are you suggesting separated DTO with just keycloak fields or with status DTO fields plus keycloak fields?

Reply:
An extension of StatusDTO like we had when it was manual. allOf with an inline object should work here.

Reply:
@Ceiu 
I was under the impression that 'allOf' will generate different classes instead of inheritance. However, after exploring open API documentation, It appears we can generate inheritance using 'discriminator'.  I have updated to code to generate StatusDTO and KeycloakStatusDTO classes. However, after adding this, one additional property (discriminator property value)  coming in the response. I need to explore to avoid this additional property in the response.

Reply:
Disregard my request here. As you noted, what we should be doing is not possible with OpenAPI short of making substantial modifications to it.

What you have here that combines the DTOs is what we're going to be doing going forward.

Reply:
@Ceiu 
I had one additional query, I have already made changes to create separate DTOs using ‘discriminator’ and pushed in this PR. 

To clarify, Is your approval for combining both StatusDTO and KeycloakStatusDTO into one or for separate DTOs using ‘discriminator’?

Reply:
Combined until we find something better.